tools/ctl/ctl.go

```diff
@@ -216,6 +217,94 @@ var getResults = &cobra.Command{
 	},
 }
 
+var missingAttestations = &cobra.Command{
+	Use:   "missing -project <ID> -run <ID> -attestation-bucket <name> -format <list|bench>",
+	Short: "Find successes from a smoketest run that don't exist in the attestation bucket.",
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		log.Default().SetOutput(cmd.ErrOrStderr())
+		var passing []firestore.Rebuild
+		{
+			req, err := buildFetchRebuildRequest(cmd.Context(), "", *runFlag, "", *clean)
+			if err != nil {
+				log.Fatal(err)
+			}
+			fireClient, err := firestore.NewClient(cmd.Context(), *project)
+			if err != nil {
+				log.Fatal(err)
+			}
+			rebuilds, err := fireClient.FetchRebuilds(cmd.Context(), req)
+			if err != nil {
+				log.Fatal(err)
+			}
+			for _, r := range rebuilds {
+				if r.Success {
```

Comment:
you should be able to do this using buildFetchRebuildRequest. If you change the filter param to a ptr to string, you can provide an empty string which will select all successes.

Reply:
Currently filter is a prefix, so I don't think changing to a string ptr would be enough. We would also have to special case the empty string to be an exact match. I also don't think filtering for an empty string is as readable as selecting based on r.Success.

We could maybe change filter to be a function type instead but I think that's probably outside the scope of this change.